v2/dtos/addressable.go

```diff
@@ -0,0 +1,85 @@
+package dtos
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/v2/models"
+)
+
+type Addressable struct {
+	common.Versionable `json:",inline"`
+
+	RESTAddressable `json:",inline"`
+	MQTTAddressable `json:",inline"`
+}
+
+type BaseAddressable struct {
+	Type string `json:"type" validate:"oneof='REST' 'MQTT'"`
+
+	Host string `json:"host"`
+	Port int    `json:"port"`
+}
+
+type RESTAddressable struct {
+	BaseAddressable `json:",inline"`
+	Path            string `json:"path,omitempty"`
+	QueryParameters string `json:"queryParameters,omitempty"`
+	HTTPMethod      string `json:"httpMethod" validate:"required,oneof='GET' 'HEAD' 'POST' 'PUT' 'DELETE' 'TRACE' 'CONNECT'"`
+}
+
+type MQTTAddressable struct {
+	BaseAddressable `json:",inline"`
+	Publisher       string `json:"publisher"`
+	Topic           string `json:"topic"`
+	QoS             int    `json:"qos,omitempty"`
```

Comment:
There are other MQTT options that we may need (ignore secure connection for now). They are:
        KeepAlive    // # Seconds (must be 2 or greater)
        Retained     // Boolean
        AutoReconnect  // Boolean
        ConnectTimeout = // Seconds

Reply:
Added.